File: UKParliament.CodeTest.Web/ClientApp/src/app/features/person-manager/models/person-view-model.ts
import { DepartmentViewModel } from "./department-view-model";

/**
 * The PersonViewModel interface.
 */
export interface PersonViewModel {
  /** Gets or sets the identifier. */
  id: number;

  /** Gets or sets the first name. */
  firstName: string;

  /** Gets or sets the last name. */
  lastName: string;

  /** Gets or sets the date of birth. */
  dateOfBirth: string; // DateOnly maps best to string in TypeScript (ISO 8601 format recommended).

  /** Gets or sets the department identifier. */
  departmentId: number;

  /** Gets or sets the department details. */
  department?: DepartmentViewModel;
}

File: UKParliament.CodeTest.ViewModels/PersonViewModel.cs
namespace UKParliament.CodeTest.ViewModels;

/// <summary>
/// The PersonViewModel class.
/// </summary>
public class PersonViewModel
{
    /// <summary>
    /// Gets or sets the identifier.
    /// </summary>
    /// <value>
    /// The identifier.
    /// </value>
    public int Id { get; set; }

    public string FirstName { get; set; }

    public string LastName { get; set; }

    public DateOnly DateOfBirth { get; set; }

    public int DepartmentId { get; set; }

    public DepartmentViewModel? Department { get; set; }
}
